### PR TITLE
Update flow and flow-typed dependencies

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

- Bumps `flow` version from  `0.59.0` to `0.64.0`.
- Bumps `flow-typed` version from `2.1.5` to `2.2.3`.
- Updates type definitions in `static/src/javascripts/__flow__/flow-typed` so they match our current set of dependencies.
- Fixes some new `flow` validation issues, predominantly around `window.setTimeout` now returning a type of `TimeoutID` and `window.setInterval` returning a type of `IntervalID`.

## What is the value of this and can you measure success?

Brings `flow` dependencies up to date.

## Does this affect other platforms - Amp, Apps, etc?

No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

No

## Screenshots

N/A

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts/projects/commercial-control/modules/hosted/youtube.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ export const initHostedYoutube = (el: HTMLElement): void => {
                         );
                     }, 1000);
                 } else {
-                    clearTimeout(playTimer);
+                    window.clearInterval(playTimer);
```
Comment (GHaberis) [inline static/src/javascripts/projects/commercial-control/modules/hosted/youtube.js]: We were previously passing `playTimer` to `clearTimeout` however `playTimer` is an `IntervalID` so we should be using `clearInterval`.

File: static/src/javascripts/projects/commercial/modules/hosted/youtube.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ export const initHostedYoutube = (el: HTMLElement): void => {
                         );
                     }, 1000);
                 } else {
-                    clearTimeout(playTimer);
+                    window.clearInterval(playTimer);
```
Comment (GHaberis) [inline static/src/javascripts/projects/commercial/modules/hosted/youtube.js]: As before, we were previously passing `playTimer` to `clearTimeout` however `playTimer` is an `IntervalID` so we should be using `clearInterval`.

File: static/src/javascripts/projects/commercial-control/modules/hosted/youtube.js (hunk 1 above)
Comment (sndrs) [reply static/src/javascripts/projects/commercial-control/modules/hosted/youtube.js]: nice!

**Comment by GHaberis** [comment]
The tests are failing as the `raven` definition from `flow-typed` is using a syntax for exports that's unsupported as of `flow` version `0.63`. 

https://github.com/facebook/flow/blob/master/Changelog.md#0630

The the flow-typed definition fir Raven will need updating, however in the meantime I'm going to see if I can replace it with a stub which returns `any`.

**Review by SiAdcock — APPROVED** [review]
Nice nice!

**Pull request merged**